diff --git a/packages/directive/package.json b/packages/directive/package.json
index e173a465bd8ab4e3f0809b38db2457da38efbbc7..cb3d0fed2903bf24f04b495374e593e26291c5d7 100644
--- a/packages/directive/package.json
+++ b/packages/directive/package.json
@@ -30,6 +30,7 @@
   },
   "files": [
     "dist",
+    "src",
     "changelog.md",
     "readme.md"
   ],
diff --git a/packages/directive/src/utils.ts b/packages/directive/src/utils.ts
index ff866dbbcf1a57a4fb6f89608bd2a80f418580cf..c4bdabd5fbb6ab112d2b8a7841265e49207560e9 100644
--- a/packages/directive/src/utils.ts
+++ b/packages/directive/src/utils.ts
@@ -7,11 +7,11 @@ import type { DirectiveLevel } from './types.js'
 export function getDirectivePattern(level: DirectiveLevel, marker: string) {
   switch (level) {
     case 'container':
-      return `^${marker}([\\s\\S]*?)\\n${marker}`
+      return String.raw`^${marker}([\s\S]*?)\n${marker}`
     case 'block':
-      return `^${marker}((?:[a-zA-Z][\\w-]*|[\\{\\[].*?[\\}\\]])+)`
+      return String.raw`^${marker}\s*((?:[a-zA-Z][\w-]*)\s*(\[[^\]]*\])?\s*(\{[^}]*\})?)`
     case 'inline':
-      return `^${marker}((?:[a-zA-Z][\\w-]*|[\\{].*?[\\}]+|[\\[].*?[\\]])+)`
+      return String.raw`^${marker}((?:[a-zA-Z][\w-]*)(\[[^\]]*\])(\{[^}]*\})?)`
   }
 }
 
diff --git a/packages/directive/vite.config.ts b/packages/directive/vite.config.ts
index e48f25ed703292f451195230a78496186c0c67a8..4ecad12a059952017cd167cb1c8a524f15912baf 100644
--- a/packages/directive/vite.config.ts
+++ b/packages/directive/vite.config.ts
@@ -12,6 +12,7 @@ export default defineConfig({
       formats: ['es', 'cjs', 'umd'],
       fileName: 'index'
     },
+    minify: false,
     rollupOptions: {
       external: ['marked', 'attribute-parser', 'moo'],
       output: {
@@ -20,7 +21,8 @@ export default defineConfig({
           moo: 'moo'
         }
       }
-    }
+    },
+    sourcemap: 'inline',
   },
   test: {
     globals: true,
